lettuce/src/main/scala/com/avast/sst/lettuce/LettuceModule.scala

```diff
@@ -14,7 +14,7 @@ object LettuceModule {
 
   /** Makes [[io.lettuce.core.RedisClient]] initialized with the given config and optionally [[io.lettuce.core.resource.ClientResources]]. */
   def makeClient[F[_]: Sync](config: LettuceConfig, clientResources: Option[ClientResources] = None): Resource[F, RedisClient] = {
-    val create = clientResources match {
+    lazy val create = clientResources match {
```

Comment:
You might as well inline the body of `create` where it's used -- it's just one place. Then there could be no danger of side effects happening at the wrong time/place.